### PR TITLE
Migrate to Circle CI and Update Dependencies

### PR DESCRIPTION
**Pull request by jturkel**
This PR migrates builds to CircleCI and updates some of the dependencies to fix build issues:

- Dropped support for Avro < 1.9.
- Dropped support for Ruby < 2.6 since Ruby 2.5 will be EOL next month.
- Added Ruby 3.0 support.
- Upgraded Rubocop

Rubocop flagged a missing call to `super` in `Avro::Builder::Types::RecordType#initialize` but fixing that requires a bit more work so I'm going to defer it for now.

### CODE REVIEW

File: .ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-2.5.8
+2.6.5
```
Comment (will89) [inline .ruby-version:1]: Should this be 2.6.6 now?

**Review by will89 — APPROVED** [review]

File: .ruby-version (hunk 1 above)
Comment (jturkel) [reply .ruby-version:1]: Oops. Yes.

**Comment by tjwp** [comment]
@jturkel Thanks for updating this. I can rebase my `json` fix once it is merged. Should the next release be v0.18 or should this go to v1.0?

**Comment by jturkel** [comment]
I think at this point we should declare the gem stable and go to 1.0. Are there any other breaking changes you think we should include in that release?

**Comment by tjwp** [comment]
> I think at this point we should declare the gem stable and go to 1.0. Are there any other breaking changes you think we should include in that release?

👍 Nothing top of mind. I can release v1.0 after getting in the pending `json` fix.

**Pull request merged**